alacritty/src/window.rs

```diff
@@ -392,7 +393,7 @@ impl Window {
 }
 
 #[cfg(not(any(target_os = "macos", windows)))]
-fn x_embed_window(window: &GlutinWindow, parent_id: u64) {
+fn x_embed_window(window: &GlutinWindow, parent_id: c_ulong) {
```

Comment:
Wouldn't a `usize` work just as well without having to rely on non-rust datatypes?

Reply:
Unfortunately no, a usize the same size as a pointer type, and longs are not guaranteed to be the same size as pointers. It needs to match the XLib C ABI exactly.

Reply:
Make a PR ;)

Reply:
It should probably happen eventually, but it's a much bigger change. In the meantime this will fix any compilation issues.

Reply:
> Unfortunately no, a usize the same size as a pointer type, and longs are not guaranteed to be the same size as pointers.

c_ulong is defined as the same size as poniters in Rust.

Reply:
Yes, practically speaking you're correct. I don't know of any other LLVM platforms besides windows where the size of pointer and long are different. So that would likely work, but isn't strictly correct as far as linking to the C ABI goes. I preferred to be closer to that since the state of the x11 crate right now takes no pains to hide that it's a thin wrapper around the XLib C interface.